packages/plugin-meta-extractor/src/webpack-plugin.ts

```diff
@@ -0,0 +1,35 @@
+import path from 'node:path';
+import * as webpack from 'webpack';
+import { extractPluginMeta } from './meta-extractor.js';
+
+const DEFAULT_GENERATED_FILENAME = 'plugin.generated.json';
+
+export type GrafanaPluginMetaExtractorOptions = {
+  entryFile?: string;
+  outputAssetName?: string;
+};
+
+export class GrafanaPluginMetaExtractor {
+  options: GrafanaPluginMetaExtractorOptions;
+
+  constructor(options: GrafanaPluginMetaExtractorOptions) {
+    this.options = options;
+  }
+
+  apply(compiler: webpack.Compiler) {
+    compiler.hooks.emit.tapAsync('GrafanaPluginMetaExtractor', (compilation, callback) => {
```

Comment:
I think `emit` is called before writing assets to the output dir. I'm thinking it's highly unlikely but... is it possible this plugin could clobber a file with the same output name?

Reply:
I am not sure yet to be honest, will double-check, thanks!

Reply:
(Resolving this as in the end we are updating the `plugin.json` file.)